crates/wysiwyg/src/composer_model/hyperlinks.rs

```diff
@@ -84,21 +83,29 @@ where
 
     fn is_blank_selection(&self, range: Range) -> bool {
         for leaf in range.leaves() {
-            if leaf.kind == LineBreak {
-                continue;
-            } else if leaf.kind == Text {
-                let text_node = self
-                    .state
-                    .dom
-                    .lookup_node(&leaf.node_handle)
-                    .as_text()
-                    .unwrap();
-                let selection_range = leaf.start_offset..leaf.end_offset;
-                if !text_node.is_blank_in_range(selection_range) {
-                    return false;
+            match leaf.kind {
+                DomNodeKind::Text => {
+                    let text_node = self
+                        .state
+                        .dom
+                        .lookup_node(&leaf.node_handle)
+                        .as_text()
+                        .unwrap();
+                    let selection_range = leaf.start_offset..leaf.end_offset;
+                    if !text_node.is_blank_in_range(selection_range) {
+                        return false;
+                    }
                 }
-            } else {
-                return false;
+                DomNodeKind::LineBreak => continue,
+                DomNodeKind::Formatting(_)
+                | DomNodeKind::Link
+                | DomNodeKind::ListItem
+                | DomNodeKind::List
+                | DomNodeKind::CodeBlock
+                | DomNodeKind::Quote
+                | DomNodeKind::Generic
+                | DomNodeKind::Mention
+                | DomNodeKind::Paragraph => return false,
```

Comment:
I added this exhaustive match in case we added any leaf nodes that could contain blank content in the future - so that it forces us to handle that. But I suppose that is pretty unlikely so we could revert this back to using an else branch.

Reply:
Maybe this should be a panic rather than `return false` if it is a container kind though.

Reply:
https://github.com/matrix-org/matrix-rich-text-editor/pull/699/commits/d4ef5c771ebc0495a63161525040127020983891